src/math/mod.rs

```diff
@@ -29,6 +29,124 @@ pub fn canon_egcd(a: i64, b: i64, c: i64) -> Option<(i64, i64, i64)> {
     }
 }
 
+fn pos_mod(n: i64, m: i64) -> i64 {
```

Comment:
We could leave this modular math stuff here temporarily until Rust gets const generics; when it does, this functionality will be duplicated by the [Field](https://github.com/EbTech/rust-algorithms/blob/master/src/math/num.rs#L174) type